File: encrypt.py
import sys


def encrypt(originalText, keyValue, line, text):
    try:
        # Read a text file character by character
        for ch in line:
            if isExeption(ch):
                text += ch
                # Upper case
            elif ch.isupper():
                ch = ch.lower()
                newCh = str(encode(ch, keyValue))
                newCh = newCh.upper()
                text += newCh
            else:
                newCh = str(encode(ch, keyValue))
                text += newCh
        # Recursive method reading line by line
        return encrypt(originalText, keyValue, originalText.next(), text)
    # EOF
    except StopIteration:
        encryptedText.write(text)
        encryptedText.close()


def encode(ch, keyValue):
    # Define a latin alphabet
    alphabet = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "l", "m",
                "n", "o", "p", "q", "r", "s", "t", "u", "v", "x", "y", "z"]
    # Search in the alphabet the same word
    for i in alphabet:
        if ch == i:
            # Encode the current character
            return alphabet[(alphabet.index(i) + keyValue) % 23]


def isExeption(ch):
    return (ch == " " or ch == "," or ch == "." or ch == "\n" or ch == "?"
            or ch == "!" or ch == ":")


if __name__ == '__main__':
    originalText = open(sys.argv[2], 'r')
    encryptedText = open(sys.argv[3], 'w')
    keyValue = int(sys.argv[1])
    text = ''
    encrypt(originalText, keyValue, originalText.readline(), text)

File: encrypt.c
#include<stdio.h>
#include<ctype.h>
#include <stdbool.h>


void encrypt(int keyValue, FILE *text, FILE *encryptedText);
char encode(char ch, int keyValue);
bool isExeption(char ch);

int main(int argc, char* argv[])
{

  int keyValue;
  FILE *text;
  FILE *encryptedText;

  if (argc != 4)
  {
    printf("Invalid number of arguments\n");
    exit(0);
  }
  else
  {
    keyValue = atoi(argv[1]);
    text = fopen(argv[2], "r");
    encryptedText = fopen(argv[3], "w");
  }
  encrypt(keyValue, text, encryptedText);
}


void encrypt(int keyValue,FILE *text, FILE *encryptedText)
{
  int x;
  char newCh;
  // Read a text file character by character
  while (( x = fgetc(text)) != EOF)
  {
    if(isExeption(x)){
      fprintf(encryptedText, "%c", x );
    }
    // Upper Case
    else if (isupper(x))
    {
      x = tolower(x);
      newCh = (char) encode(x, keyValue);
      newCh = toupper(newCh);
      fprintf(encryptedText, "%c", newCh );

    }
    else
    {
    newCh =  (char) encode(x, keyValue);
    fprintf(encryptedText, "%c", newCh );

    }
  }
  fclose(text);

}

char encode(char ch, int keyValue)
{
  //Define a latin alphabet
  char *alphabet_a[11] = {'b', 'd', 'f', 'h', 'l','n', 'p', 'r', 't', 'v', 'y'};

  char *alphabet_b[12] = {'a', 'c', 'e', 'g', 'i', 'm','o', 'q', 's', 'u', 'x', 'z'};

  //Search in the alphabet the same word
  for(int i = 0; i < 12; i++)
  {
    if(ch == alphabet_b[i])
    {
        return alphabet_b[(i + (keyValue*2)) % 11];
    }else if (ch== alphabet_a[i]) {
        return alphabet_a[(i+keyValue % 10)];
    }
  }
}

bool isExeption(char ch)
{
  return ch == ' ' || ch == ',' || ch == '.' || ch == '?' || ch == '!' ||
  ch == '"' || ch == '\n' || ch == ';' || ch == ':' || ch == ']' || ch == '[';

}
